### ISSUE TITLE
Add support for username/pwd in URL

### ISSUE THREAD

**Issue by floragunncom** [floragunncom/search-guard-kibana-plugin#37]
See here: https://groups.google.com/d/msgid/search-guard/b9431977-d3f3-40b2-bc0f-9f18241765ba%40googlegroups.com?utm_medium=email&utm_source=footer

**Comment by yasvanth** [comment]
Hi Team,
Is there any temporary workaround to pass the username and password in login.html in the location 
<pre>
/kibana/plugins/searchguard/public/apps/login
<pre>

Because this machine is not accessible to outside .

Thnaks,
Yash

**Comment by floragunncom** [comment]
No, you'd need to patch the plugin. But if you know what you're doing, you can change the title or subtitle on the login page and display a default user/password there:

http://floragunncom.github.io/search-guard-docs/kibana.html

But be aware that nearly 60% of all attacks come from the inside, not the outside. Therefore, we absolutely do not recommend to rely on a machine not being accessible from the outside!

**Comment by yasvanth** [comment]
Thanks for your valuable suggestion. I have created a casper.js to do auto login ( temporary) . Works as expected. But I wont carry this workaround as long term.
Will wait for next update for the kibana plugin.

Thanks,
Yash

**Comment by Pommes9485** [comment]
I recently introduced the Kibana Plugin and Searchguard to my system as well and were very frustrated when I discovered, that this feature is missing. Would you be so kind to share your workaround with me?

**Comment by yasvanth** [comment]
@Pommes9485 The workaround is very simple which I used casper.js(Slimerjs engine) to auto login to basic authentication in kibana . Below link will help you to do basic authentication .
[http://code-epicenter.com/how-to-login-to-amazon-using-casperjs-working-example/](url)

once you logged in you can use your dashboar url to hit after authentication succeeded.

If you are looking for casperjs for sso login let me know will share a script.

Thanks,
Yash

**Comment by floragunncom** [comment]
Pls. note that all major browsers have stopped supporting the user:pass@host syntax, for obvious security reasons. We also think that supporting it (officially) would open up all kinds of security leaks. The current recommendation is to use (SSO with) JWT. They can be attached as an URL parameter.

**Comment by Pommes9485** [comment]
@yasvanth
Thanks for the reply,  to be honest, i didnt think i would get one. 
I was about to do something similiar in our application, whilst not knowing, that there is already stuff for that.
It would be great, if you could elaborate a bit more. Where did you put the js File, which will be executed per server request? Or did you save it locally? 
It would be perfect, if you could send me your code, although the implementation did not seem that hard. 

@floragunncom
I know that. I really do. Unfortunatly, the plugin is not usable for our application, if there is no way to automatically login an user. Kibana will be integrated into the application and another login screen after our own login is neither acceptable nor professional.

**Comment by yasvanth** [comment]
Hi @Pommes9485 ,
Sorry for the late response.  The script (casper js headless browser )authenticates to IDP and receives the cookies which is stored in local system. And loads the Dasboad URL. The Auto login script is located in client side not on kibana side. 

Below I have provided the code. If you are using form authentication , the scrip will change.

<pre>
var casper = require('casper').create({
  verbose: true,
  logLevel: "debug",
  waitTimeout: 10000,
  viewportSize: { width: 1920, height: 1080 },
  pageSettings: {
    javascriptEnabled: true,
    loadImages: true,
    resourceTimeout: 150000,
    loadPlugins: true,
    webSecurityEnabled: false
  }
});

var mouse = require("mouse").create(casper);
var f = require("utils").format;
var url = "kibana url";
var utils = require('utils');
var x = require('casper').selectXPath;
var idp = 'Authentication_backend IDP ';
var username_field = 'j_username';
var password_field = 'j_password';
var login_string = {};
      login_string[username_field] = '<UserName>';
      login_string[password_field] = '<Password>';
var form_str = 'form';
var login_timeout =3000;
var repeat = 'forever';

casper.start('<sso_Authentication_URL>');
casper.then(function() {
   casper.waitForSelector(form_str, function() {
         var loginform   = this.evaluate(function() {
            return document.querySelector('form');
         });

         var myloginform = casper.evaluate(function(intloginform) {
            var input = document.createElement('input');
            input.type = 'hidden';
            input.name = '_eventId_proceed';
            input.value = 'flower';
            document.forms[0].appendChild(input);
            return document.forms[0];
         }, loginform );
// Fill up the form and authenticate
      this.fill(form_str, login_string, true);
   }, function timeout() {
      fail("Login failed!, authentication timeout");
   },login_timeout);
});



casper.then(function() {
        casper.thenOpen('<Dashboard_Url>');
        casper.then(function(){
                casper.wait(18000, function(){
                        this.capture('/var/html/www/web/screen.png');
                });
        });
});

function timerize(p) {
    perf_data.push(p);
}

   casper.run(function() {this.exit();});

</pre>

Thanks,
Yash

**Comment by jochenkressin** [comment]
@mgustafsson1 I think we can implement this relatively easy now on the new codebase, right? However, I still think this opens potential security leaks. Passwords in URLs is not something we would like to promote. So I'm willing to implement it, but users need to enable the feature actively by flipping a config switch in kibana.yml. The default should be the behavior as we have it now (no support for username/passwords in URLs).

**Comment by Bader972016** [comment]
Hey, any updates on this feature for SearchGuard 6.6.0?
